netket/experimental/operator/_fermion_operator_2nd.py

```diff
@@ -248,10 +248,15 @@ def copy(self, *, dtype: Optional[DType] = None):
         op._operators = dict(zip(terms, weights))
         return op
 
-    def reduce(self):
+    def _reduce(self):
         """Reduce the number of operators by bringing them to normal form and grouping equivalent terms"""
-        self._operators = _reduce_operators(self._operators, self.dtype)
-        return self
+        op_dict = _reduce_operators(self._operators, self.dtype)
+        terms = list(op_dict.keys())
+        weights = list(op_dict.values())
+        op = FermionOperator2nd(
+            self.hilbert, terms, weights, constant=self._constant, dtype=self.dtype
+        )
+        return op
```

Comment:
Wether this is inplace/outofplace I believe should depend on what is the use case you envisage:
 - if it should be run automatically internally by the `_setup` method, then it should be inplace as it's just an internal method that you use to setup the operator. In that case it should be marked as internal.
 - If it is very expensive, and you expect that most people don't need it, you can make it part of the public API and then it should be out of place.

My vague feeling here is that we should go for option (1), but I don't have an understanding of how expensive that is, so..

Reply:
My goal was still to remove this again, since it might blow up. Cleaning up unnecessary zeros is already sufficient to keep things light for now..